core/src/main/java/org/modelmapper/internal/ProxyFactory.java

```diff
@@ -62,15 +64,57 @@
       throw errors.invocationAgainstFinalClass(type).toException();
 
     try {
-      return OBJENESIS.newInstance(new ByteBuddy()
+      final DynamicType.Unloaded<T> unloaded = new ByteBuddy()
           .subclass(type)
           .method(METHOD_FILTER)
           .intercept(InvocationHandlerAdapter.of(interceptor))
-          .make()
-          .load(useOSGiClassLoaderBridging ? BridgeClassLoaderFactory.getClassLoader(type) : type.getClassLoader())
-          .getLoaded());
+          .make();
+      final ClassLoadingStrategy<ClassLoader> classLoadingStrategy = chooseClassLoadingStrategy(type);
+      if (classLoadingStrategy != null) {
+        return OBJENESIS.newInstance(unloaded
+            .load(useOSGiClassLoaderBridging ? BridgeClassLoaderFactory.getClassLoader(type) : type.getClassLoader(), classLoadingStrategy)
+            .getLoaded());
+      } else {
+        return OBJENESIS.newInstance(unloaded
+            .load(useOSGiClassLoaderBridging ? BridgeClassLoaderFactory.getClassLoader(type) : type.getClassLoader())
+            .getLoaded());
+      }
     } catch (Throwable t) {
       throw errors.errorInstantiatingProxy(type, t).toException();
     }
   }
-}
\ No newline at end of file
+
+  private static Method privateLookupIn;
+  private static Object lookup;
+
+  static {
+    try {
+      Class<?> methodHandles = Class.forName("java.lang.invoke.MethodHandles");
+      lookup = methodHandles.getMethod("lookup").invoke(null);
+      privateLookupIn = methodHandles.getMethod(
+          "privateLookupIn",
+          Class.class,
+          Class.forName("java.lang.invoke.MethodHandles$Lookup")
+      );
+    } catch (Exception e) {
+      // ignore
+    }
+  }
+
+  private static <T> ClassLoadingStrategy<ClassLoader> chooseClassLoadingStrategy(Class<T> type) {
+    try {
+      final ClassLoadingStrategy<ClassLoader> strategy;
+      if (ClassInjector.UsingLookup.isAvailable() && privateLookupIn != null && lookup != null) {
+        Object privateLookup = privateLookupIn.invoke(null, type, lookup);
+        strategy = ClassLoadingStrategy.UsingLookup.of(privateLookup);
+      } else if (ClassInjector.UsingReflection.isAvailable()) {
+        strategy = ClassLoadingStrategy.Default.INJECTION;
+      } else {
+        throw new IllegalStateException("No code generation strategy available");
+      }
+      return strategy;
+    } catch (Exception e) {
+      return null;
```

Comment:
I think we don't need to catch the Exception and return null.
We should just throw the IllegalStateException and let the caller to handle the exception.